roles/keystone/tasks/main.yml

```diff
@@ -31,9 +31,49 @@
     - etype: other
       permission: r
 
-- name: install keystone service
-  upstart_service: name=keystone user=keystone cmd=/usr/local/bin/keystone-all
-                   config_dirs=/etc/keystone
+- name: install keystone uwsgi service
+  template: src=roles/keystone/templates/etc/init/keystone.conf
+            dest=/etc/init/keystone.conf mode=0644
+  with_items:
+    - { uwsgi_path: "{{ openstack_source.virtualenv_base }}/keystone/bin/uwsgi" }
+  when: openstack_install_method == 'source'
+  
+- name: install keystone uwsgi service
+  template: src=roles/keystone/templates/etc/init/keystone.conf
+            dest=/etc/init/keystone.conf mode=0644
+  with_items:
+    - { uwsgi_path: "{{ openstack_package.virtualenv_base }}/keystone/bin/uwsgi" }
+  when: openstack_install_method == 'package'
+
+- name: Creates keystone uwsgi directory
+  file: path=/etc/keystone/uwsgi state=directory
+        owner=keystone group=keystone mode=0775
+
+- name: Creates keystone httpd uwsgi directory
+  file: path=/var/www/keystone state=directory
+        owner=keystone group=keystone mode=0775
+
+- name: copy keystone httpd file to admin location
+  command: cp /opt/stack/keystone/httpd/keystone.py /var/www/keystone/admin
```

Comment:
`command` with cp will fail on a second run, because the file will already be there. It'll also fail to copy over a newer version of the file.

I would suggest instead using the `synchronize` module, as discussed in  https://github.com/ansible/ansible/issues/9982